applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_d_plus_d_minus_damage.cpp

```diff
@@ -586,5 +587,16 @@ template class GenericSmallStrainDplusDminusDamage<GenericTensionConstitutiveLaw
 template class GenericSmallStrainDplusDminusDamage<GenericTensionConstitutiveLawIntegratorDplusDminusDamage<DruckerPragerYieldSurface<VonMisesPlasticPotential<6>>>, GenericCompressionConstitutiveLawIntegratorDplusDminusDamage<VonMisesYieldSurface<VonMisesPlasticPotential<6>>>>;
 template class GenericSmallStrainDplusDminusDamage<GenericTensionConstitutiveLawIntegratorDplusDminusDamage<DruckerPragerYieldSurface<VonMisesPlasticPotential<6>>>, GenericCompressionConstitutiveLawIntegratorDplusDminusDamage<TrescaYieldSurface<VonMisesPlasticPotential<6>>>>;
 template class GenericSmallStrainDplusDminusDamage<GenericTensionConstitutiveLawIntegratorDplusDminusDamage<DruckerPragerYieldSurface<VonMisesPlasticPotential<6>>>, GenericCompressionConstitutiveLawIntegratorDplusDminusDamage<DruckerPragerYieldSurface<VonMisesPlasticPotential<6>>>>;
+template class GenericSmallStrainDplusDminusDamage<GenericTensionConstitutiveLawIntegratorDplusDminusDamage<MohrCoulombYieldSurface<VonMisesPlasticPotential<6>>>, GenericCompressionConstitutiveLawIntegratorDplusDminusDamage<MohrCoulombYieldSurface<VonMisesPlasticPotential<6>>>>;

```

Comment:
We need to find the way to do more automatic these combinations. Maybe using some macros magic @roigcarlo ?

Reply:
yep that is sth we have to discuss
the number of those combinations is increasing exponentially and I fear that the linking-time will kill us eventually :/

Reply:
The problem is going to increase in near future because some people wants to use these CL in 2D so we will have to register all these combinations in 2D!!!

Reply:
oh ...
it should probably discussed then what is the best solution for this (maybe polymorphism (runtime) instead of templating (compiletime), also then the registration of the combinations would not be needed). Otherwise increasing the compilation / linking time will affect many developers

Still I am not the expert here, @roigcarlo could you comment pls?

@RiccardoRossi maybe we can speak in a larger group what the mid/long-term goals are for this

Reply:
Maybe here is a good idea that this class in particular is header only.